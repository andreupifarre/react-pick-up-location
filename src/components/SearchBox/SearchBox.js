import React from 'react';
import './SearchBox.scss';

const SearchField = () => <></>;

const SearchBox = () => (
  <div className="c-search-box">
    <h2 className="c-search-box__title">Letâ€™s find your ideal car</h2>
    <SearchField />
  </div>
);

export default SearchBox;
