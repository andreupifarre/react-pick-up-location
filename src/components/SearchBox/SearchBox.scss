.c-search-box {
  margin: 0;
  padding: 8px;
  letter-spacing: -0.4px;
  background-color: #f3ce56;
  box-sizing: border-box;
  max-width: 480px;

  &__title {
    margin: 0 0 15px;
    font-family: Ubuntu, Open Sans, Segoe UI, Tahoma, sans-serif;
    font-size: 24px;
    font-weight: 500;
    line-height: 2rem;
  }
}

@media (min-width: 30em) {
  .c-search-box {
    padding: 24px;
    letter-spacing: 0;
  }
}

@media (min-width: 64em) {
  .c-searchbox {
    width: 480px;
  }
}
