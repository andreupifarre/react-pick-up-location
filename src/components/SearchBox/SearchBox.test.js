import React from 'react';
import { shallow } from 'enzyme';
import SearchBox from './SearchBox';

const wrapper = shallow(<SearchBox />);

describe('SearchBox', () => {
  it('should contain an element with a class .c-search-box', () => {
    expect(wrapper.find('.c-search-box').exists()).toBeTruthy();
  });

  it('should contain an H2 element with a class .c-search-box__title', () => {
    const element = wrapper.find('.c-search-box__title');
    expect(element.exists()).toBeTruthy();
    expect(element.name()).toEqual('h2');
  });

  it('should contain a SearchField component', () => {
    expect(wrapper.find('SearchField').exists()).toBeTruthy();
  });

  it('should render correctly', () => {
    expect(wrapper.getElements()).toMatchSnapshot();
  });
});
