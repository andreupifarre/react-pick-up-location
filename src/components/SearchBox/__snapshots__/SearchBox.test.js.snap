// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox should render correctly 1`] = `
Array [
  <div
    className="c-search-box"
  >
    <h2
      className="c-search-box__title"
    >
      Letâ€™s find your ideal car
    </h2>
    <SearchField />
  </div>,
]
`;
